cms/djangoapps/contentstore/views/utilities/bulkupdate.py

```diff
@@ -0,0 +1,245 @@
+            'course_outline_url': course_outline_url,
+            'bulkupdate_update_url': update_url,
+            'bulkupdate_status_url': status_url,
+            'default_max_attempts': DEFAULT_MAX_ATTEMPTS,
+            'default_show_answer': DEFAULT_SHOW_ANSWER,
+            'show_answer_options': SHOW_ANSWER_OPTIONS
+        }
+    )
+
+
+def _utility_bulkupdate_post_handler(request, course_key_string):
+    """
+    Internal bulkupdate handler for POST operation
+    """
+    modified_settings = {}
+    user = request.user
+
+    # Retrieve settings from request
+    try:
+        max_attempts = request.POST.get('maxAttempts')
+        show_answer = request.POST.get('showAnswer')
+        session_status = request.session.setdefault("update_status", {})
+        session_status_string = course_key_string + max_attempts + show_answer
+    except:
+        return JsonResponse(
+            {
+                'ErrMsg': 'Missing or invalid arguments',
+                'Stage': -1
+            },
+            status=400
+        )
+
+    # Validate settings
+    _save_request_status(request, session_status_string, 1)
+
+    if max_attempts.find('null') == -1:
+        try:
+            max_attempts = int(max_attempts)
+        except:
+            _save_request_status(request, session_status_string, -1)
+            return JsonResponse(
+                {
+                    'ErrMsg': 'Given value for max attempts is not an integer',
+                    'Stage': -1
+                },
+                status=400
+            )
+        if max_attempts < 0:
+            _save_request_status(request, session_status_string, -1)
+            return JsonResponse(
+                {
+                    'ErrMsg': 'Given value for max attempts is negative',
+                    'Stage': -1
+                },
+                status=400
+            )
+        else:
+            modified_settings['max_attempts'] = max_attempts
+
+    if show_answer.find('null') == -1:
+        if show_answer in SHOW_ANSWER_OPTIONS:
+            modified_settings['showanswer'] = show_answer
+        else:
+            _save_request_status(request, session_status_string, -1)
+            return JsonResponse(
+                {
+                    'ErrMsg': 'Given value for show answer is not an available option',
+                    'Stage': -1
+                },
+                status=400
+            )
+    _save_request_status(request, session_status_string, 2)
+    _update_advanced_settings(request, course_key_string, session_status_string, modified_settings)
+    _save_request_status(request, session_status_string, 3)
+    bulk_update_problem_settings.delay(course_key_string, user.id, user.username, user.email, modified_settings)
+
+    return JsonResponse({'Status': 'OK'})
+
+
+@login_required
+def utility_bulkupdate_handler(request, course_key_string):
+    """
+    Handler for bulk update requests in the utilities tool. Currently
```

Comment:
Docstrings are used in a number of "magic" ways in python.
There are some benefits to formatting them like a commit message and/or email:

```
"""
One sentence sound byte

Further explanation here.
"""
```

Reply:
done